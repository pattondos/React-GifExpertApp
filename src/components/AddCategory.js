import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const hadleInputChange = event => {
        setInputValue(event.target.value);
    }

    const handleSubmit = event => {
        event.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(categorie => [inputValue, ...categorie]);
            setInputValue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add Category</h2>
            <input
                type="text"
                value={inputValue}
                onChange={hadleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
