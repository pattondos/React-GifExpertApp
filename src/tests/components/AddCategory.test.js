import { shallow } from 'enzyme';
import React from 'react';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe mostrarse correctamente ', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola prro';

        input.simulate('change', { target: { value } });

        // expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('NO debe enviar la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('Debe de llamar setCategories y limpiar la caja de texto', () => {

        const value = 'Hola';
        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        // expect(setCategories).toHaveBeenCalledTimes(1);
        // expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');

    });



})
