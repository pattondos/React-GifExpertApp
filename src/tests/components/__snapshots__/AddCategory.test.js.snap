// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
